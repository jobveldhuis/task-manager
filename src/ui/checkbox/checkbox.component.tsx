import { GestureResponderEvent, Pressable, StyleSheet } from "react-native";
import { COLORS } from "@/util/colors.const";
import { Icon } from "../icon";

type CheckboxProps = {
  isChecked: boolean;
  onPress: (event: GestureResponderEvent) => void;
  onLongPress: (event: GestureResponderEvent) => void;
  testID?: string;
};

export function Checkbox({
  isChecked,
  onPress: handlePress,
  onLongPress: handleLongPress,
  testID,
}: CheckboxProps): JSX.Element {
  return (
    <Pressable
      style={[styles.base, isChecked && styles.checked]}
      onPress={handlePress}
      onLongPress={handleLongPress}
      testID={testID}
    >
      {isChecked && <Icon name="check" variant="inverse" />}
    </Pressable>
  );
}

const styles = StyleSheet.create({
  base: {
    width: 28,
    height: 28,
    justifyContent: "center",
    alignItems: "center",
    borderRadius: 4,
    borderColor: COLORS.primary,
    borderWidth: 2,
  },
  checked: {
    borderWidth: 0,
    backgroundColor: COLORS.button.primary,
  },
});
