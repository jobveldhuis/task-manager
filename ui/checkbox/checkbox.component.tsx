import { GestureResponderEvent, Pressable, StyleSheet } from "react-native";
import { Icon } from "@/ui/icon";
import { COLORS } from "@/util/colors.const";

type CheckboxProps = {
  isChecked: boolean;
  onPress: (event: GestureResponderEvent) => void;
};

export function Checkbox({
  isChecked,
  onPress: handlePress,
}: CheckboxProps): JSX.Element {
  return (
    <Pressable
      style={[styles.base, isChecked && styles.checked]}
      onPress={handlePress}
    >
      {isChecked && <Icon name="check" variant="inverse" />}
    </Pressable>
  );
}

const styles = StyleSheet.create({
  base: {
    width: 28,
    height: 28,
    justifyContent: "center",
    alignItems: "center",
    borderRadius: 4,
    borderColor: COLORS.primary,
    borderWidth: 2,
  },
  checked: {
    borderWidth: 0,
    backgroundColor: COLORS.button.primary,
  },
});
