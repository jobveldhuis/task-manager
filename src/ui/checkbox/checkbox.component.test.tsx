import { Checkbox } from "@/ui/checkbox/checkbox.component";
import { fireEvent, render, screen } from "@testing-library/react-native";

describe("Checkbox", () => {
  it("should match snapshot", () => {
    expect(
      render(
        <Checkbox isChecked onPress={() => {}} onLongPress={() => {}} />,
      ).toJSON(),
    ).toMatchSnapshot();
  });

  it("should handle a press event", () => {
    const handlePress = jest.fn();
    const handleLongPress = jest.fn();
    render(
      <Checkbox
        isChecked
        onPress={handlePress}
        onLongPress={handleLongPress}
        testID="checkbox"
      />,
    );

    const checkbox = screen.getByTestId("checkbox");
    expect(checkbox).toBeTruthy();
    fireEvent(checkbox, "press");
    expect(handlePress).toHaveBeenCalled();
    expect(handleLongPress).not.toHaveBeenCalled();
  });

  it("should handle a long press event", () => {
    const handlePress = jest.fn();
    const handleLongPress = jest.fn();
    render(
      <Checkbox
        isChecked
        onPress={handlePress}
        onLongPress={handleLongPress}
        testID="checkbox"
      />,
    );

    const checkbox = screen.getByTestId("checkbox");
    expect(checkbox).toBeTruthy();
    fireEvent(checkbox, "longPress");
    expect(handlePress).not.toHaveBeenCalled();
    expect(handleLongPress).toHaveBeenCalled();
  });
});
