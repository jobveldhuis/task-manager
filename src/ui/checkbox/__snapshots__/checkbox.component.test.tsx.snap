// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should match snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "borderColor": "#000C49",
        "borderRadius": 4,
        "borderWidth": 2,
        "height": 28,
        "justifyContent": "center",
        "width": 28,
      },
      {
        "backgroundColor": "#343C64",
        "borderWidth": 0,
      },
    ]
  }
>
  <Text />
</View>
`;
